test/filters/test_experimental_voxel_grid.cpp

```diff
@@ -0,0 +1,387 @@
+  new_grid.setLeafSize(0.05f, 0.05f, 0.05f);
+  old_grid.setLeafSize(0.05f, 0.05f, 0.05f);
+  new_grid.setInputCloud(cloud_rgb);
+  old_grid.setInputCloud(cloud_rgb);
+
+  new_grid.setDownsampleAllData(false);
+  old_grid.setDownsampleAllData(false);
+  new_grid.filter(new_out);
+  old_grid.filter(old_out);
+  EXPECT_POINTS_EQ(new_out, old_out);
+  new_out.clear();
+  old_out.clear();
+
+  new_grid.setDownsampleAllData(true);
+  old_grid.setDownsampleAllData(true);
+  new_grid.filter(new_out);
+  old_grid.filter(old_out);
+  EXPECT_POINTS_EQ(new_out, old_out);
+  new_out.clear();
+  old_out.clear();
+
+  new_grid.setMinimumPointsNumberPerVoxel(5);
+  old_grid.setMinimumPointsNumberPerVoxel(5);
+  new_grid.filter(new_out);
+  old_grid.filter(old_out);
+  EXPECT_POINTS_EQ(new_out, old_out);
+}
+
+TEST(PointXYZRGBA, ExperimentalVoxelGridEquivalency)
+{
+  PointCloud<PointXYZRGBA> new_out, old_out;
+
+  pcl::experimental::VoxelGrid<PointXYZRGBA> new_grid;
+  pcl::VoxelGrid<PointXYZRGBA> old_grid;
+  new_grid.setLeafSize(0.05f, 0.05f, 0.05f);
+  old_grid.setLeafSize(0.05f, 0.05f, 0.05f);
+  new_grid.setInputCloud(cloud_rgba);
+  old_grid.setInputCloud(cloud_rgba);
+
+  new_grid.setDownsampleAllData(false);
+  old_grid.setDownsampleAllData(false);
+  new_grid.filter(new_out);
+  old_grid.filter(old_out);
+  EXPECT_POINTS_EQ(new_out, old_out);
+  new_out.clear();
+  old_out.clear();
+
+  new_grid.setDownsampleAllData(true);
+  old_grid.setDownsampleAllData(true);
+  new_grid.filter(new_out);
+  old_grid.filter(old_out);
+  EXPECT_POINTS_EQ(new_out, old_out);
+  new_out.clear();
+  old_out.clear();
+
+  new_grid.setMinimumPointsNumberPerVoxel(5);
+  old_grid.setMinimumPointsNumberPerVoxel(5);
+  new_grid.filter(new_out);
+  old_grid.filter(old_out);
+  EXPECT_POINTS_EQ(new_out, old_out);
+}
+
+int
+main(int argc, char** argv)
+{
+  // Load a standard PCD file from disk
+  if (argc < 4) {
+    std::cerr << "No test file given. Please download `bun0.pcd`, "
+                 "`milk_cartoon_all_small_clorox.pcd` and `milk_color.pcd`, and pass "
+                 "their paths to the test."
+              << std::endl;
+    return (-1);
+  }
+
+  // Load a standard PCD file from disk
+  loadPCDFile(argv[1], *cloud);
+  loadPCDFile(argv[2], *cloud_rgb);
+
+  // Input
+  if (loadPCDFile(argv[1], *cloud) < 0) {
+    std::cerr << "Failed to read test file. Please download `bun0.pcd` and pass its "
+                 "path to the test."
+              << std::endl;
+    return (-1);
+  }
+  if (loadPCDFile(argv[2], *cloud_rgb) < 0) {
+    std::cerr << "Failed to read test file. Please download "
+                 "`milk_cartoon_all_small_clorox.pcd` and pass its path to the test."
+              << std::endl;
+    return (-1);
+  }
+  if (loadPCDFile(argv[3], *cloud_rgba) < 0) {
+    std::cerr << "Failed to read test file. Please download `milk_color.pcd` and pass "
+                 "its path to the test."
+              << std::endl;
+    return (-1);
+  }
```

Comment:
You load the files before checking for errors. Mistake?

Reply:
Yes they are my mistake, those two lines are deleted now